File: ios/Runner/Extension/SongModelExtension.swift
//
//  SongModelExtension.swift
//  Runner
//
//  Created by t k on 2024/08/14.
//

extension SongModel {
    func toPigeon() -> SongDTO {
        return SongDTO(
            id: String(self.id.raw),
            name: self.name,
            albumId: String(self.albumId.raw),
            albumName: self.albumName,
            artistId: String(self.artistId.raw),
            artistName: self.artistName,
            imageId: self.imageId,
            duration: self.duration,
            trackNumber: Double(self.trackNumber)
        )
    }
}

File: android/app/src/main/kotlin/dev/tcode/thinmpf/extension/SongModelExtension.kt
package dev.tcode.thinmpf.extension

import dev.tcode.thinmpf.model.contract.SongModelContract
import dev.tcode.thinmpf.pigeon_output.SongDTO

fun SongModelContract.toPigeon(): SongDTO {
    return SongDTO(
        id = this.id.raw,
        name = this.name,
        albumId = this.albumId.raw,
        albumName = this.albumName,
        artistId = this.artistId.raw,
        artistName = this.artistName,
        imageId = this.imageId,
        duration = this.duration.toDouble() / 1000,
        trackNumber = this.trackNumber
    )
}
